src/_pytest/compat.py

```diff
@@ -35,13 +36,20 @@
 
 if TYPE_CHECKING:
     from typing import Type  # noqa: F401 (used in type string)
+    from typing_extensions import Final
 
 
 _T = TypeVar("_T")
 _S = TypeVar("_S")
 
 
-NOTSET = object()
```

Comment:
`object()` doesn't produce a singleton on the type level, so I had to switch it to an idiom that does.